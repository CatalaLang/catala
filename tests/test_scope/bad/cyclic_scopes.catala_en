## Article

```catala
declaration scope A:
  context b scope B
  context x content integer

declaration scope B:
  context a scope A
  context y content integer

scope A:
  definition x equals b.y

scope B:
  definition y equals a.x
```
