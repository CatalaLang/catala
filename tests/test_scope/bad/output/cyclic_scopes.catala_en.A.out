[ERROR] Cyclic dependency detected between scopes!
[ERROR] 
[ERROR] Cycle variable B, declared:
[ERROR]  --> test_scope/bad/cyclic_scopes.catala_en
[ERROR]   | 
[ERROR] 8 | declaration scope B:
[ERROR]   |                   ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle variable A:
[ERROR]  --> test_scope/bad/cyclic_scopes.catala_en
[ERROR]   | 
[ERROR] 5 |   context b scope B
[ERROR]   |           ^
[ERROR]   + Article
[ERROR] 
[ERROR] Cycle variable A, declared:
[ERROR]  --> test_scope/bad/cyclic_scopes.catala_en
[ERROR]   | 
[ERROR] 4 | declaration scope A:
[ERROR]   |                   ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle variable B:
[ERROR]  --> test_scope/bad/cyclic_scopes.catala_en
[ERROR]   | 
[ERROR] 9 |   context a scope A
[ERROR]   |           ^
[ERROR]   + Article
